### DISCUSSION TITLE
Add the property to ignore dependency tree crawling (experimental) - for dynamic imports

### DISCUSSION THREAD

**Discussion by Shperung** [lingui/js-lingui#1910] [Ideas]
I'm currently utilizing dependency tree crawling for my project, as described in the Lingui documentation here:  https://lingui.dev/guides/message-extraction#dependency-tree-crawling-experimental
In my TypeScript files (.tsx), I have two types of imports: static imports like

`import SectionModel from '../../source/section/section.model';` 
and dynamic imports
 `const getProductsHitBrandMethod = (
      await import('./methods/getProductsHitBrand-provider-method')
    )`
    
For my use case, I don't require dependency tree crawling for files with dynamic imports. Therefore, I'd like to propose adding an option to ignore dependency tree crawling (experimental) for files containing dynamic imports in JavaScript.

Please share your thoughts and feedback on this idea.

**Comment by timofei-iatsenko** [comment]
This is too specific case which is most likely will not be needed for other people. The deps-extractor uses esbuild under the hood to crawl the import tree. The options for esbuild are exposed in lingui config. I believe there is no ready to use option as well, but you can implement your own esbuild plugin and ignore files you don't need
